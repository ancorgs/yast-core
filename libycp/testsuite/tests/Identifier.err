Parsed:
----------------------------------------------------------------------
{
  module "test";
  export this_is_global;
  integer `a = 1;
  global integer `this_is_global = 2;
}
----------------------------------------------------------------------
[Interpreter] tests/Identifier.ycp:18 Builtin 'export this_is_global;' not implemented
Parsed:
----------------------------------------------------------------------
{
  integer `a = 55;
  return [test::this_is_global, ::a, a];
}
----------------------------------------------------------------------
[Interpreter] tests/Identifier.ycp:29 [YCP] Undeclared variable ::a
Parsed:
----------------------------------------------------------------------
{
  global integer `a = 55;
  return {
  integer `a = 42;
  return [test::this_is_global, ::a, a];
}
;
}
----------------------------------------------------------------------
[Interpreter] tests/Identifier.ycp:38 Local symbol 'a' shadows global symbol
Parsed:
----------------------------------------------------------------------
{
  integer `a = W::a;
  integer `b = S::Read (x, y);
  integer `c = UI::OpenDialog ();
  integer `d = UI::``{
  return OpenDialog ();
}
;
  integer `d1 = WFM::test::a;
  integer `d2 = WFM::test::b ();
  integer `d3 = WFM::test::c (x);
  integer `d4 = WFM::test::d (x, y);
  return 0;
}
----------------------------------------------------------------------
[Interpreter] tests/Identifier.ycp:45 [YCP] Undeclared variable W::a
[Interpreter] tests/Identifier.ycp:45 Local symbol 'a' shadows global symbol
[Interpreter] tests/Identifier.ycp:46 [YCP] Undeclared variable x
[Interpreter] tests/Identifier.ycp:46 [YCP] Undeclared variable y
[Interpreter] tests/Identifier.ycp:46 Undefined function S::Read()
[libycp] YCPBasicInterpreter.cc(evaluateUI):96 dummy evaluateUI (OpenDialog ())
[libycp] YCPBasicInterpreter.cc(evaluateUI):96 dummy evaluateUI (``{
  return OpenDialog ();
}
)
[libycp] YCPBasicInterpreter.cc(evaluateWFM):90 dummy evaluateWFM (test::a)
[libycp] YCPBasicInterpreter.cc(evaluateWFM):90 dummy evaluateWFM (test::b ())
[libycp] YCPBasicInterpreter.cc(evaluateWFM):90 dummy evaluateWFM (test::c (x))
[libycp] YCPBasicInterpreter.cc(evaluateWFM):90 dummy evaluateWFM (test::d (x, y))
