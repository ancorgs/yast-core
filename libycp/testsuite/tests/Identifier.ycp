
# ---------------------------------------------------------
#
#  Filename:    Identifier.ycp
#
#  Purpose:     test identifier ycp type
#
#  Creator:     kkaempf@suse.de
#
#  Maintainer:  kkaempf@suse.de
#
# ---------------------------------------------------------


{
  module "test";

  export this_is_global;

  integer a = 1;
  global integer this_is_global = 2;
  // implicit: return nil;
}

{
  integer a = 55;

  // should be [2, nil, 55]
  return [ test::this_is_global, ::a, a ];
}

{
  global integer a = 55;	// this is accessible via ::a

  // should be [2, 55, 42]
  //   and a warning about local a shadowing global a

  return { integer a = 42; return [ test::this_is_global, ::a, a ]; };
}

{
  // *** This is just a syntactical test ***
  // This code only has semantical value when executed in WFM

  integer a = W::a;
  integer b = S::Read(x,y);
  integer c = UI::OpenDialog();
  integer d = UI::``{ return OpenDialog();};
  integer d1 = WFM::test::a;
  integer d2 = WFM::test::b();
  integer d3 = WFM::test::c(x);
  integer d4 = WFM::test::d(x,y);
  return 0;
}
