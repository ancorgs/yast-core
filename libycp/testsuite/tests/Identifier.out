(nil)
([2, nil, 55])
([2, 55, 42])
(0)
